oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/VersionGarbageCollector.java

```diff
@@ -1048,8 +1061,9 @@ public void collectGarbage(final NodeDocument doc, final GCPhases phases) {
                     updateOpList.add(op);
                 }
             }
-            if (log.isDebugEnabled()) {
-                log.debug("UpdateOp for {} is {}", doc.getId(), op);
+            if (DETAILED_GC_LOG.isDebugEnabled() && op.hasChanges()) {
+                // only log in case of changes & debug level enabled
+                DETAILED_GC_LOG.debug("UpdateOp for {} is {}", doc.getId(), op);
```

Comment:
Not sure we should have the entire UpdateOp in the audit log - it can be very noisy. Also not sure if logging the property names is fine here... (we could move it to log.trace, perhaps, not sure..)